In [2]:
#importing the necessary python libraries
import pandas as pd
import numpy as np

In [3]:
#reading the wickets csv file
df = pd.read_csv("wickets.csv", encoding = 'unicode_escape')

In [4]:
#displaying the first 10 rows of the dataframe
display(df.head(10))

Player       Span   Mat  Inns  Balls   Runs  Wkts    BBI  \
0  M Muralitharan (ICC/SL)  1992-2010   133   230  44039  18180   800   9/51   
1           SK Warne (AUS)  1992-2007   145   273  40705  17995   708   8/71   
2        JM Anderson (ENG)  2003-2021  164*   304  35079  16575   623   7/42   
3         A Kumble (INDIA)  1990-2008   132   236  40850  18355   619  10/74   
4         GD McGrath (AUS)  1993-2007   124   243  29248  12186   563   8/24   
5          SCJ Broad (ENG)  2007-2021   149   274  29863  14590   524   8/15   
6            CA Walsh (WI)  1984-2001   132   242  30019  12688   519   7/37   
7            DW Steyn (SA)  2004-2019    93   171  18608  10077   439   7/51   
8      N Kapil Dev (INDIA)  1978-1994   131   227  27740  12867   434   9/83   
9        HMRKB Herath (SL)  1999-2018    93   170  25993  12157   433  9/127   

      BBM    Ave  Econ    SR   5  10  
0  16/220  22.72  2.47  55.0  67  22  
1  12/128  25.41  2.65  57.4  37  10  
2   11/71  26.60  2.83  56.3  30   3  
3  14/149  29.65  2.69  65.9  35   8  
4   10/27  21.64  2.49  51.9  29   3  
5  11/121  27.84  2.93  56.9  18   3  
6   13/55  24.44  2.53  57.8  22   3  
7   11/60  22.95  3.24  42.3  26   5  
8  11/146  29.64  2.78  63.9  23   2  
9  14/184  28.07  2.80  60.0  34   9

<b>Player</b> : Name of the players who has bowled in test cricket<br> 
<b>Span</b> : Duration of time the player was active in playing cricket<br> 
<b>Mat</b> : Number of Matches the player has played<br> 
<b>Inns</b> : Number of innings the player has bowled in<br> 
<b>Balls</b> : Number of Balls the player has bowled<br> 
<b>Runs</b> : Number of runs the player has conceded<br> 
<b>BBI</b> : Best Bowling figure in an Innings<br> 
<b>BBM</b> : Best Bowling figure in a Match<br> 
<b>Ave</b> : Average which is calculated by dividing the numbers of runs conceded by the number of wickets taken<br> 
<b>Econ</b> : Economy Rate, which is the average number of runs conceded for each over bowled<br> 
<b>SR</b> : Strike Rate, which is calculated as total balls bowled / wickets taken by bowler<br> 
<b>5</b> : Number of times the player has taken 5 wickets in a match<br> 
<b>10</b> : Number of times the player has taken 10 wickets in a match<br> 

In [5]:
#displaying the number of rows and columns in the dataframe
display(df.shape)

(79, 14)

In [6]:
#displaying the data statistics
display(df.describe())

Inns         Balls          Runs        Wkts        Ave  \
count   79.000000     79.000000     79.000000   79.000000  79.000000   
mean   144.911392  18638.354430   8599.354430  317.215190  27.469747   
std     51.180222   7199.256972   3085.168807  121.924911   3.655658   
min     67.000000   8785.000000   4846.000000  200.000000  20.940000   
25%    110.000000  13583.000000   6456.500000  229.000000  24.500000   
50%    129.000000  16498.000000   7742.000000  266.000000  28.000000   
75%    169.000000  21742.500000   9756.000000  374.500000  29.870000   
max    304.000000  44039.000000  18355.000000  800.000000  34.790000   

            Econ         SR          5         10  
count  79.000000  79.000000  79.000000  79.000000  
mean    2.806835  59.193671  16.354430   2.797468  
std     0.351577   9.350132   9.642372   3.235935  
min     1.980000  41.200000   3.000000   0.000000  
25%     2.600000  53.300000   9.500000   1.000000  
50%     2.820000  57.400000  14.000000   2.000000  
75%     3.080000  63.950000  20.500000   3.500000  
max     3.460000  91.900000  67.000000  22.000000

In [7]:
#displaying the data types
display(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 79 entries, 0 to 78
Data columns (total 14 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Player  79 non-null     object 
 1   Span    79 non-null     object 
 2   Mat     79 non-null     object 
 3   Inns    79 non-null     int64  
 4   Balls   79 non-null     int64  
 5   Runs    79 non-null     int64  
 6   Wkts    79 non-null     int64  
 7   BBI     79 non-null     object 
 8   BBM     79 non-null     object 
 9   Ave     79 non-null     float64
 10  Econ    79 non-null     float64
 11  SR      79 non-null     float64
 12  5       79 non-null     int64  
 13  10      79 non-null     int64  
dtypes: float64(3), int64(6), object(5)
memory usage: 8.8+ KB


None

**Observation** <br>
There are no missing values in the dataset

In [8]:
#renaming the column names appropriately
df = df.rename(columns={"Mat":"Match", "Inns":"Innings","Wkts":"Wickets","BBI":"Best Bowling figure/Innings","BBM":"Best Bowling figure/Match","Ave":"Average","Econ":"Economy","SR":"Strike Rate", "5":"5 wickets/match","10":"10 wickets/match"})

display(df.head())

Player       Span Match  Innings  Balls   Runs  Wickets  \
0  M Muralitharan (ICC/SL)  1992-2010   133      230  44039  18180      800   
1           SK Warne (AUS)  1992-2007   145      273  40705  17995      708   
2        JM Anderson (ENG)  2003-2021  164*      304  35079  16575      623   
3         A Kumble (INDIA)  1990-2008   132      236  40850  18355      619   
4         GD McGrath (AUS)  1993-2007   124      243  29248  12186      563   

  Best Bowling figure/Innings Best Bowling figure/Match  Average  Economy  \
0                        9/51                    16/220    22.72     2.47   
1                        8/71                    12/128    25.41     2.65   
2                        7/42                     11/71    26.60     2.83   
3                       10/74                    14/149    29.65     2.69   
4                        8/24                     10/27    21.64     2.49   

   Strike Rate  5 wickets/match  10 wickets/match  
0         55.0               67                22  
1         57.4               37                10  
2         56.3               30                 3  
3         65.9               35                 8  
4         51.9               29                 3

In [9]:
#removing columns named 'Best Bowling figure/Innings' and 'Best Bowling figure/Match' from the dataframe
df = df.drop('Best Bowling figure/Innings', axis = 1)
df = df.drop('Best Bowling figure/Match', axis = 1)

display(df.head())

Player       Span Match  Innings  Balls   Runs  Wickets  \
0  M Muralitharan (ICC/SL)  1992-2010   133      230  44039  18180      800   
1           SK Warne (AUS)  1992-2007   145      273  40705  17995      708   
2        JM Anderson (ENG)  2003-2021  164*      304  35079  16575      623   
3         A Kumble (INDIA)  1990-2008   132      236  40850  18355      619   
4         GD McGrath (AUS)  1993-2007   124      243  29248  12186      563   

   Average  Economy  Strike Rate  5 wickets/match  10 wickets/match  
0    22.72     2.47         55.0               67                22  
1    25.41     2.65         57.4               37                10  
2    26.60     2.83         56.3               30                 3  
3    29.65     2.69         65.9               35                 8  
4    21.64     2.49         51.9               29                 3

In [10]:
#splitting the country part from the player's name in the 'Player' column
df_player = df['Player'].str.split("(", expand = True) 

display(df_player.head())

0        1
0  M Muralitharan   ICC/SL)
1        SK Warne      AUS)
2     JM Anderson      ENG)
3        A Kumble    INDIA)
4      GD McGrath      AUS)

In [11]:
df = pd.concat([df, df_player], axis = 1)

display(df.head())

Player       Span Match  Innings  Balls   Runs  Wickets  \
0  M Muralitharan (ICC/SL)  1992-2010   133      230  44039  18180      800   
1           SK Warne (AUS)  1992-2007   145      273  40705  17995      708   
2        JM Anderson (ENG)  2003-2021  164*      304  35079  16575      623   
3         A Kumble (INDIA)  1990-2008   132      236  40850  18355      619   
4         GD McGrath (AUS)  1993-2007   124      243  29248  12186      563   

   Average  Economy  Strike Rate  5 wickets/match  10 wickets/match  \
0    22.72     2.47         55.0               67                22   
1    25.41     2.65         57.4               37                10   
2    26.60     2.83         56.3               30                 3   
3    29.65     2.69         65.9               35                 8   
4    21.64     2.49         51.9               29                 3   

                 0        1  
0  M Muralitharan   ICC/SL)  
1        SK Warne      AUS)  
2     JM Anderson      ENG)  
3        A Kumble    INDIA)  
4      GD McGrath      AUS)

In [12]:
df = df.drop('Player', axis = 1)
display(df.head())

Span Match  Innings  Balls   Runs  Wickets  Average  Economy  \
0  1992-2010   133      230  44039  18180      800    22.72     2.47   
1  1992-2007   145      273  40705  17995      708    25.41     2.65   
2  2003-2021  164*      304  35079  16575      623    26.60     2.83   
3  1990-2008   132      236  40850  18355      619    29.65     2.69   
4  1993-2007   124      243  29248  12186      563    21.64     2.49   

   Strike Rate  5 wickets/match  10 wickets/match                0        1  
0         55.0               67                22  M Muralitharan   ICC/SL)  
1         57.4               37                10        SK Warne      AUS)  
2         56.3               30                 3     JM Anderson      ENG)  
3         65.9               35                 8        A Kumble    INDIA)  
4         51.9               29                 3      GD McGrath      AUS)

In [13]:
df = df.rename(columns={0:'Player',1:'Country'})
display(df.head())

Span Match  Innings  Balls   Runs  Wickets  Average  Economy  \
0  1992-2010   133      230  44039  18180      800    22.72     2.47   
1  1992-2007   145      273  40705  17995      708    25.41     2.65   
2  2003-2021  164*      304  35079  16575      623    26.60     2.83   
3  1990-2008   132      236  40850  18355      619    29.65     2.69   
4  1993-2007   124      243  29248  12186      563    21.64     2.49   

   Strike Rate  5 wickets/match  10 wickets/match           Player  Country  
0         55.0               67                22  M Muralitharan   ICC/SL)  
1         57.4               37                10        SK Warne      AUS)  
2         56.3               30                 3     JM Anderson      ENG)  
3         65.9               35                 8        A Kumble    INDIA)  
4         51.9               29                 3      GD McGrath      AUS)

In [14]:
df['Country'] = df['Country'].str.replace(")","")
display(df.head())

<ipython-input-14-78515716fddb>:1: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will*not* be treated as literal strings when regex=True.
  df['Country'] = df['Country'].str.replace(")","")


Span Match  Innings  Balls   Runs  Wickets  Average  Economy  \
0  1992-2010   133      230  44039  18180      800    22.72     2.47   
1  1992-2007   145      273  40705  17995      708    25.41     2.65   
2  2003-2021  164*      304  35079  16575      623    26.60     2.83   
3  1990-2008   132      236  40850  18355      619    29.65     2.69   
4  1993-2007   124      243  29248  12186      563    21.64     2.49   

   Strike Rate  5 wickets/match  10 wickets/match           Player Country  
0         55.0               67                22  M Muralitharan   ICC/SL  
1         57.4               37                10        SK Warne      AUS  
2         56.3               30                 3     JM Anderson      ENG  
3         65.9               35                 8        A Kumble    INDIA  
4         51.9               29                 3      GD McGrath      AUS

In [15]:
new_col_sequence = ['Player', 'Country','Span','Match','Innings','Balls','Runs','Wickets','Average','Economy','Strike Rate','5 wickets/match','10 wickets/match']

df = df[new_col_sequence]
display(df.head())

Player Country       Span Match  Innings  Balls   Runs  Wickets  \
0  M Muralitharan   ICC/SL  1992-2010   133      230  44039  18180      800   
1        SK Warne      AUS  1992-2007   145      273  40705  17995      708   
2     JM Anderson      ENG  2003-2021  164*      304  35079  16575      623   
3        A Kumble    INDIA  1990-2008   132      236  40850  18355      619   
4      GD McGrath      AUS  1993-2007   124      243  29248  12186      563   

   Average  Economy  Strike Rate  5 wickets/match  10 wickets/match  
0    22.72     2.47         55.0               67                22  
1    25.41     2.65         57.4               37                10  
2    26.60     2.83         56.3               30                 3  
3    29.65     2.69         65.9               35                 8  
4    21.64     2.49         51.9               29                 3

In [16]:
#function to check if 'ICC' is in the 'Country' column
def icc_check(x):
    if "ICC" in x:
        return 1
    else:
        return 0

In [17]:
df['Played for ICC'] = df['Country'].apply(icc_check)
display(df.head())

Player Country       Span Match  Innings  Balls   Runs  Wickets  \
0  M Muralitharan   ICC/SL  1992-2010   133      230  44039  18180      800   
1        SK Warne      AUS  1992-2007   145      273  40705  17995      708   
2     JM Anderson      ENG  2003-2021  164*      304  35079  16575      623   
3        A Kumble    INDIA  1990-2008   132      236  40850  18355      619   
4      GD McGrath      AUS  1993-2007   124      243  29248  12186      563   

   Average  Economy  Strike Rate  5 wickets/match  10 wickets/match  \
0    22.72     2.47         55.0               67                22   
1    25.41     2.65         57.4               37                10   
2    26.60     2.83         56.3               30                 3   
3    29.65     2.69         65.9               35                 8   
4    21.64     2.49         51.9               29                 3   

   Played for ICC  
0               1  
1               0  
2               0  
3               0  
4               0

In [18]:
df['Played for ICC'].value_counts()

0    74
1     5
Name: Played for ICC, dtype: int64

**Observation** <br>
It can be deduced that 5 players played for ICC.


In [19]:
df['Country'] = df['Country'].str.replace("ICC/","")
df['Country'] = df['Country'].str.replace("/ICC","")
display(df.head())

Player Country       Span Match  Innings  Balls   Runs  Wickets  \
0  M Muralitharan       SL  1992-2010   133      230  44039  18180      800   
1        SK Warne      AUS  1992-2007   145      273  40705  17995      708   
2     JM Anderson      ENG  2003-2021  164*      304  35079  16575      623   
3        A Kumble    INDIA  1990-2008   132      236  40850  18355      619   
4      GD McGrath      AUS  1993-2007   124      243  29248  12186      563   

   Average  Economy  Strike Rate  5 wickets/match  10 wickets/match  \
0    22.72     2.47         55.0               67                22   
1    25.41     2.65         57.4               37                10   
2    26.60     2.83         56.3               30                 3   
3    29.65     2.69         65.9               35                 8   
4    21.64     2.49         51.9               29                 3   

   Played for ICC  
0               1  
1               0  
2               0  
3               0  
4               0

In [20]:
df['Country'].value_counts()

AUS      18
ENG      15
INDIA    10
WI        9
SA        8
PAK       7
NZ        7
SL        3
BDESH     1
ZIM       1
Name: Country, dtype: int64

**Observations** <br>
1. There are 10 different countries present in this dataset. <br>
2. There are 18 Australian Bowlers present in this dataset. <br>
3. There is only 1 Bangladeshi player present in this dataset.

In [21]:
#slicing the start and end years from the 'Span' column
df['start_year'] = df['Span'].str[0:4]
df['end_year'] = df['Span'].str[5:]
display(df.head())

Player Country       Span Match  Innings  Balls   Runs  Wickets  \
0  M Muralitharan       SL  1992-2010   133      230  44039  18180      800   
1        SK Warne      AUS  1992-2007   145      273  40705  17995      708   
2     JM Anderson      ENG  2003-2021  164*      304  35079  16575      623   
3        A Kumble    INDIA  1990-2008   132      236  40850  18355      619   
4      GD McGrath      AUS  1993-2007   124      243  29248  12186      563   

   Average  Economy  Strike Rate  5 wickets/match  10 wickets/match  \
0    22.72     2.47         55.0               67                22   
1    25.41     2.65         57.4               37                10   
2    26.60     2.83         56.3               30                 3   
3    29.65     2.69         65.9               35                 8   
4    21.64     2.49         51.9               29                 3   

   Played for ICC start_year end_year  
0               1       1992     2010  
1               0       1992     2007  
2               0       2003     2021  
3               0       1990     2008  
4               0       1993     2007

In [22]:
#converting the 'start_year' and 'end_year' column into integer data type
df['start_year'] = df['start_year'].astype('int')
df['end_year'] = df['end_year'].astype('int')


In [23]:
#calculating the years played by a player
df['Years Played'] = df['end_year'] - df['start_year']
display(df.head())

Player Country       Span Match  Innings  Balls   Runs  Wickets  \
0  M Muralitharan       SL  1992-2010   133      230  44039  18180      800   
1        SK Warne      AUS  1992-2007   145      273  40705  17995      708   
2     JM Anderson      ENG  2003-2021  164*      304  35079  16575      623   
3        A Kumble    INDIA  1990-2008   132      236  40850  18355      619   
4      GD McGrath      AUS  1993-2007   124      243  29248  12186      563   

   Average  Economy  Strike Rate  5 wickets/match  10 wickets/match  \
0    22.72     2.47         55.0               67                22   
1    25.41     2.65         57.4               37                10   
2    26.60     2.83         56.3               30                 3   
3    29.65     2.69         65.9               35                 8   
4    21.64     2.49         51.9               29                 3   

   Played for ICC  start_year  end_year  Years Played  
0               1        1992      2010            18  
1               0        1992      2007            15  
2               0        2003      2021            18  
3               0        1990      2008            18  
4               0        1993      2007            14

In [24]:
#removing the 'start_year' and 'end_year' columns from the dataframe
df = df.drop('start_year', axis = 1)
df = df.drop('end_year', axis = 1)

display(df.head())

Player Country       Span Match  Innings  Balls   Runs  Wickets  \
0  M Muralitharan       SL  1992-2010   133      230  44039  18180      800   
1        SK Warne      AUS  1992-2007   145      273  40705  17995      708   
2     JM Anderson      ENG  2003-2021  164*      304  35079  16575      623   
3        A Kumble    INDIA  1990-2008   132      236  40850  18355      619   
4      GD McGrath      AUS  1993-2007   124      243  29248  12186      563   

   Average  Economy  Strike Rate  5 wickets/match  10 wickets/match  \
0    22.72     2.47         55.0               67                22   
1    25.41     2.65         57.4               37                10   
2    26.60     2.83         56.3               30                 3   
3    29.65     2.69         65.9               35                 8   
4    21.64     2.49         51.9               29                 3   

   Played for ICC  Years Played  
0               1            18  
1               0            15  
2               0            18  
3               0            18  
4               0            14

In [25]:
#sorting players with respect to years they played
df = df.sort_values(by="Years Played", ascending = False)
display(df.head())
display(df.tail())

Player Country       Span Match  Innings  Balls   Runs  Wickets  \
21      Imran Khan      PAK  1971-1992    88      142  19458   8258      362   
55       GS Sobers       WI  1954-1974    93      159  21599   7999      235   
9     HMRKB Herath       SL  1999-2018    93      170  25993  12157      433   
0   M Muralitharan       SL  1992-2010   133      230  44039  18180      800   
2      JM Anderson      ENG  2003-2021  164*      304  35079  16575      623   

    Average  Economy  Strike Rate  5 wickets/match  10 wickets/match  \
21    22.81     2.54         53.7               23                 6   
55    34.03     2.22         91.9                6                 0   
9     28.07     2.80         60.0               34                 9   
0     22.72     2.47         55.0               67                22   
2     26.60     2.83         56.3               30                 3   

    Played for ICC  Years Played  
21               0            21  
55               0            20  
9                0            19  
0                1            18  
2                0            18

Player Country       Span Match  Innings  Balls  Runs  Wickets  \
54    Yasir Shah      PAK  2014-2021   46*       84  13607  7248      235   
61   SJ Harmison      ENG  2002-2009    63      115  13375  7192      226   
72  JR Hazlewood      AUS  2014-2021    55      103  11887  5438      212   
71      K Rabada       SA  2015-2021    47       86   8785  4846      213   
44      GP Swann      ENG  2008-2013    60      109  15349  7642      255   

    Average  Economy  Strike Rate  5 wickets/match  10 wickets/match  \
54    30.84     3.19         57.9               16                 3   
61    31.82     3.22         59.1                8                 1   
72    25.65     2.74         56.0                9                 0   
71    22.75     3.30         41.2               10                 4   
44    29.96     2.98         60.1               17                 3   

    Played for ICC  Years Played  
54               0             7  
61               1             7  
72               0             7  
71               0             6  
44               0             5

**Observations** <br>
1. Imran Khan of Pakistan had played for the longest period of time i.e. 21 years <br>
2. GP Swann of England had played for the shortest period of time i.e. 5 years

In [26]:
#sorting players with respect to their Economy rate
df = df.sort_values(by="Economy")
display(df.head())

Player Country       Span Match  Innings  Balls  Runs  Wickets  \
32      LR Gibbs       WI  1958-1976    79      148  27115  8989      309   
47      R Benaud      AUS  1952-1964    63      116  19108  6704      248   
35  DL Underwood      ENG  1966-1982    86      151  21862  7674      297   
39       BS Bedi    INDIA  1966-1979    67      118  21364  7637      266   
68   CV Grimmett      AUS  1925-1936    37       67  14513  5231      216   

    Average  Economy  Strike Rate  5 wickets/match  10 wickets/match  \
32    29.09     1.98         87.7               18                 2   
47    27.03     2.10         77.0               16                 1   
35    25.83     2.10         73.6               17                 6   
39    28.71     2.14         80.3               14                 1   
68    24.21     2.16         67.1               21                 7   

    Played for ICC  Years Played  
32               0            18  
47               0            12  
35               0            16  
39               0            13  
68               0            11

**Observation** <br>
LR Gibbs of West Indies had the lowest economy rate (1.98)

In [27]:
#sorting players with respect to Strike Rate
df = df.sort_values(by="Strike Rate")
display(df.head())

Player Country       Span Match  Innings  Balls   Runs  Wickets  \
71      K Rabada       SA  2015-2021    47       86   8785   4846      213   
7       DW Steyn       SA  2004-2019    93      171  18608  10077      439   
20  Waqar Younis      PAK  1989-2003    87      154  16224   8788      373   
19   MD Marshall       WI  1978-1991    81      151  17584   7876      376   
25     AA Donald       SA  1992-2002    72      129  15519   7344      330   

    Average  Economy  Strike Rate  5 wickets/match  10 wickets/match  \
71    22.75     3.30         41.2               10                 4   
7     22.95     3.24         42.3               26                 5   
20    23.56     3.25         43.4               22                 5   
19    20.94     2.68         46.7               22                 4   
25    22.25     2.83         47.0               20                 3   

    Played for ICC  Years Played  
71               0             6  
7                0            15  
20               0            14  
19               0            13  
25               0            10

**Observation** <br>
K Rabada of South Africa had the lowest strike rate (41.2)

In [28]:
#sorting players with respect to their Average
df = df.sort_values(by="Average")
display(df.head())

Player Country       Span Match  Innings  Balls   Runs  Wickets  \
19  MD Marshall       WI  1978-1991    81      151  17584   7876      376   
41     J Garner       WI  1977-1987    58      111  13169   5433      259   
15  CEL Ambrose       WI  1988-2000    98      179  22103   8501      405   
33   FS Trueman      ENG  1952-1965    67      127  15178   6625      307   
4    GD McGrath      AUS  1993-2007   124      243  29248  12186      563   

    Average  Economy  Strike Rate  5 wickets/match  10 wickets/match  \
19    20.94     2.68         46.7               22                 4   
41    20.97     2.47         50.8                7                 0   
15    20.99     2.30         54.5               22                 3   
33    21.57     2.61         49.4               17                 3   
4     21.64     2.49         51.9               29                 3   

    Played for ICC  Years Played  
19               0            13  
41               0            10  
15               0            12  
33               0            13  
4                0            14

**Observation** <br>
MD Marshall of West Indies had the lowest bowling average (20.94)